### PR TITLE
Happy thoughts

### PR DESCRIPTION
**Pull request by IrupePozoGraviz**
API
https://irupe-happy-thoughts-api-kobjwpkrba-lz.a.run.app

happy thoughts frontend
https://joyful-nougat-d637a7.netlify.app/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -20,6 +20,85 @@ app.use(express.json());
...
+    res.status(200).json({
+      success: true,
+      response: thoughts,
+      message: "Thoughts retrieved successfully"
+    });
+  } catch(e) {
+    res.status(400).json({
+      success: false,
+      response: e,
+      message: "did not successfully"
```
Comment (JohLeo) [inline server.js:59]: You could maybe clearify this message a bit more! ;)

File: server.js [hunk 2]
```diff
@@ -20,6 +20,85 @@ app.use(express.json());
...
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0 
+
+  },
+  createdAt: {
+    type: Date,
+    default: new Date()
```
Comment (JohLeo) [inline server.js]: Did you get this code to work? Impressed, I had to change it for it to log the time accordingly. Interesting!

**Review by JohLeo — APPROVED** [review]
Well structured code and easy to follow and read!  Such a great job! Nicely done!

File: README.md [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+Technologies
+I've used Node.js, Express.js, and MongoDB. The Mongoose ODM has been particularly helpful for data modeling and validation. I've implemented POST requests for creating new thoughts and GET requests for retrieving them.
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+API https://irupe-happy-thoughts-api-kobjwpkrba-lz.a.run.app 
+
+happy thoughts frontend https://joyful-nougat-d637a7.netlify.app/
```
Comment (Nahnahke) [inline README.md:11]: Excellent read me.

File: server.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
```
Comment (Nahnahke) [inline server.js:3]: Suggest you add Schema here in order to use const Schema on line 26.

File: server.js [hunk 5]
```diff
@@ -18,8 +19,114 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
 });
+////// Tuesday ////
+
+const { Schema } = mongoose;
```
Comment (Nahnahke) [inline server.js:26]: Add Schema in the imports in order to use const { Schema } here.

File: server.js [hunk 6]
```diff
@@ -18,8 +19,114 @@ app.use(express.json());
...
+    res.status(200).json({
+      success: true,
+      response: thoughts,
+      message: "Thoughts retrieved successfully"
+    });
+  } catch(e) {
+    res.status(400).json({
+      success: false,
+      response: e,
+      message: "did not successfully"
```
Comment (Nahnahke) [inline server.js:59]: Agree with Johanna on this one. A suggestion might be something like:

"Failed to retrieve thoughts"

File: server.js [hunk 7]
```diff
@@ -18,8 +19,114 @@ app.use(express.json());
...
+           message: "created successfully"
+         });
+       } catch (e) {
+         res.status(400).json({
+          success: false,
+          response: e,
+          message: "error occured"
+        });
+      }
+   });*/
```
Comment (Nahnahke) [inline server.js:127]: I suggets you remove the commented out part of the code :)

File: server.js [hunk 8]
```diff
@@ -18,8 +19,114 @@ app.use(express.json());
...
+     } catch (e) {
+       res.status(400).json({
+        success: false,
+        response: e,
+        message: "error occured"
+      });
+    }
+ });
+
+ // POST thoughts/:thoughtId/like
```
Comment (Nahnahke) [inline server.js:83]: Some real nice work below!

File: server.js [hunk 9]
```diff
@@ -18,8 +19,114 @@ app.use(express.json());
...
+ app.post("/thoughts/:thoughtId/like", async (req, res) => {
+  try {
+    const likes = await HappyThoughts.findOneAndUpdate(
+      { "_id": req.params.thoughtId }, // Corrected parameter name
+      { $inc: { "hearts": 1 } },
+      { new: true }
+    );
+    res.status(201).json(likes);
+  } catch (err) {
+    res.status(400).json({ message: "Could not save your like", error: err.errors });
```
Comment (Nahnahke) [inline server.js:93]: What does the "error: err.errors" do? Have not used the same approach, would be interesting to know!

**Review by Nahnahke — CHANGES_REQUESTED** [review]
Would like to start out by saying this is a job well done! Your code is short and concise and very easy to follow.

I added some small suggestions, like erasing the commented out code and another suggestion might be to add more description in your code about what all the different functions do.

Great going Irro, you did a truly great job with this!

**Pull request closed without merging**